From c5b9a4be4935e531fd9fa674ba165fd40944e96a Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 17:22:27 +0800
Subject: [PATCH 268/301] AOSCOS: staging: sb105x: revise type of
 mp_chars_in_buffer() as unsigned int

Per commit fff4ef17a940 ("tty: make tty_operations::chars_in_buffer return
uint"), member function `chars_in_buffer()' in `struct tty_operations' was
revised from type `int' to type `unsigned int'.

Revise driver code accordingly.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index 593835f0ad11..8712e7423064 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -41,7 +41,7 @@ static int mp_put_char(struct tty_struct *tty, unsigned char ch);
 static void mp_put_chars(struct tty_struct *tty);
 static ssize_t mp_write(struct tty_struct *tty, const unsigned char *buf, long unsigned int count);
 static unsigned int mp_write_room(struct tty_struct *tty);
-static int mp_chars_in_buffer(struct tty_struct *tty);
+static unsigned int mp_chars_in_buffer(struct tty_struct *tty);
 static void mp_flush_buffer(struct tty_struct *tty);
 static void mp_send_xchar(struct tty_struct *tty, char ch);
 static void mp_throttle(struct tty_struct *tty);
@@ -678,7 +678,7 @@ static unsigned int mp_write_room(struct tty_struct *tty)
 	return uart_circ_chars_free(&state->info->xmit);
 }
 
-static int mp_chars_in_buffer(struct tty_struct *tty)
+static unsigned int mp_chars_in_buffer(struct tty_struct *tty)
 {
 	struct sb_uart_state *state = tty->driver_data;
 
-- 
2.48.1

